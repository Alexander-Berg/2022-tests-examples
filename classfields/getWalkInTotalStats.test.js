const MockDate = require('mockdate');
const _ = require('lodash');

const walkInMock = require('auto-core/react/dataDomain/walkIn/mocks/withData.mock');

const getWalkInTotalStats = require('./getWalkInTotalStats');

beforeEach(() => {
    MockDate.set('2019-01-05');
});

afterEach(() => {
    MockDate.reset();
});

it('должен обогащать статистику визитов лейблами, именами и цветами', () => {
    expect(getWalkInTotalStats(walkInMock)).toMatchSnapshot();
});

it('должен изменить элементы для текущего и предыдущего дня, если там нет данных', () => {
    const walkInMockClone = _.cloneDeep(walkInMock);
    walkInMockClone.walkIn.aggregation.daily_stats.points[1].date = '2019-01-05';
    walkInMockClone.walkIn.aggregation.daily_stats.points[1].visits = {};

    expect(getWalkInTotalStats(walkInMockClone)).toMatchSnapshot();
});
