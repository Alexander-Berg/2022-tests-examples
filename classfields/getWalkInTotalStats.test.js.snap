// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`должен изменить элементы для текущего и предыдущего дня, если там нет данных 1`] = `
Array [
  Object {
    "date": "2019-01-01",
    "index": 0,
    "label": "20",
    "products": Array [
      Object {
        "color": "#C1B8E8",
        "label": "20",
        "name": "Посетители",
        "sum": 20,
      },
    ],
  },
  Object {
    "customTooltip": "Данные ещё собираются и будут готовы позже.",
    "date": "2019-01-05",
    "disabled": true,
    "index": 1,
    "label": "",
    "products": Array [
      Object {
        "sum": 460,
      },
    ],
  },
]
`;

exports[`должен обогащать статистику визитов лейблами, именами и цветами 1`] = `
Array [
  Object {
    "date": "2019-01-01",
    "index": 0,
    "label": "20",
    "products": Array [
      Object {
        "color": "#C1B8E8",
        "label": "20",
        "name": "Посетители",
        "sum": 20,
      },
    ],
  },
  Object {
    "date": "2019-01-02",
    "index": 1,
    "label": "60",
    "products": Array [
      Object {
        "color": "#C1B8E8",
        "label": "60",
        "name": "Посетители",
        "sum": 60,
      },
    ],
  },
]
`;
